src/api/store/community.py

```diff
@@ -853,13 +853,23 @@ def can_use_this_subdomain(subdomain: str, community: Community=None) -> bool:
     return False
 
 def reserve_subdomain(subdomain: str, community: Community=None):
+    # first check that we can use this domain 
```

Comment:
why don't we first check whether the Subdomain exists for the community and already is in use, and if so return without doing anything.  This will be the case 99% of the time.

Reply:
It is actually the same thing that is implemented.  I can make it return early but the outcome is the same.  I just wanted to reuse the function above to handle that case.

Reply:
It doesn't really do the same thing.  Remember most of the time the cadmin will be changing something else, very rarely will the subdomain change.  So in most cases, can_use_this_subdomain will return true, then you'll mark it as unused, then you'll search for it, find it, and mark it as used.  Why not just check it first and do nothing if it exists and is used?

Reply:
the outcome is the same.  but i am happy to refactor to fail early